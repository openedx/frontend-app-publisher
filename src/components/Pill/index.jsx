import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {
  REVIEW_BY_LEGAL,
  REVIEW_BY_INTERNAL,
  PUBLISHED,
  REVIEWED,
  UNPUBLISHED,
  ARCHIVED,
} from '../../data/constants';

const Pill = ({ statuses }) => {
  const pills = [];
  statuses.forEach((status) => {
    switch (status) {
      case ARCHIVED:
        pills.push({
          text: 'Archived',
          className: 'badge badge-light',
        });
        break;
      case UNPUBLISHED:
        pills.push({
          text: 'Unsubmitted',
          className: 'badge badge-warning',
        });
        break;
      case REVIEWED:
        pills.push({
          text: 'Scheduled',
          className: 'badge badge-primary',
        });
        break;
      case REVIEW_BY_LEGAL:
      case REVIEW_BY_INTERNAL:
        pills.push({
          text: 'In review',
          className: 'badge badge-light',
        });
        pills.push({
          text: <i className="fa fa-lock" />,
        });
        break;
      case PUBLISHED:
        pills.push({
          text: 'Published',
          className: 'badge badge-success',
        });
        break;
      default:
        break;
    }
  });
  return pills.map(pill => (
    <span className={classNames('ml-2', pill.className)} key={pill.text}>
      {pill.text}
    </span>
  ));
};

Pill.propTypes = {
  statuses: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Pill;
