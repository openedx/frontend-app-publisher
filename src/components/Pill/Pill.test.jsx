import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Pill from './index';

describe('Pill', () => {
  it('renders without any statuses', () => {
    const component = shallow(<Pill statuses={[]} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders with invalid statuses', () => {
    const statuses = [undefined, null, 'fnjdsahf'];
    const component = shallow(<Pill statuses={statuses} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders with a single status', () => {
    const statuses = ['review_by_legal'];
    const component = shallow(<Pill statuses={statuses} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders with multiple statuses', () => {
    const statuses = ['review_by_internal', 'published'];
    const component = shallow(<Pill statuses={statuses} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
