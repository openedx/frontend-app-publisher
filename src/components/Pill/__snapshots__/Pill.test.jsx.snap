// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill renders with a single status 1`] = `
Array [
  <span
    className="ml-2 badge badge-light"
    key="In review"
  >
    In review
  </span>,
  <span
    className="ml-2"
    key="[object Object]"
  >
    <i
      className="fa fa-lock"
    />
  </span>,
]
`;

exports[`Pill renders with invalid statuses 1`] = `null`;

exports[`Pill renders with multiple statuses 1`] = `
Array [
  <span
    className="ml-2 badge badge-light"
    key="In review"
  >
    In review
  </span>,
  <span
    className="ml-2"
    key="[object Object]"
  >
    <i
      className="fa fa-lock"
    />
  </span>,
  <span
    className="ml-2 badge badge-success"
    key="Published"
  >
    Published
  </span>,
]
`;

exports[`Pill renders without any statuses 1`] = `null`;
